### PR TITLE
Add the ability to manager osx groups

### PR DESCRIPTION
**Pull request by adambarnett52**
Adding in adduser,deluser and members into mac_groups to give the
ability to add,delete and mange user groups on osx

### CODE REVIEW

**Comment by jfindlay** [comment]
Nice, thanks @adambarnett52.  There are some [lint errors](https://jenkins.saltstack.com/job/salt-pr-lint-n/9138/violations/file/salt/modules/mac_group.py/).  Also, will you add docstrings to the new functions?  See `salt/modules/groupadd.py` for examples.  In this case the docstrings should also include a `..versionadded:: Boron`.  Let me know if you have any questions.

**Comment by adambarnett52** [comment]
Hi @jfindlay i have addressed the issues you pointed out and added the docstrings as requested

Thanks

File: salt/modules/mac_group.py [hunk 1]
```diff
@@ -103,6 +103,71 @@ def delete(name):
...
+        salt '*' group.members foo 'user1,user2,user3,...'
+
+    Replaces a membership list for a local group 'foo'.
+    '''
+    retcode = 1
+    grp_info = __salt__['group.info'](name)
+    if grp_info and name in grp_info['name']:
+        cmd = '/usr/bin/dscl . -delete /Groups/{0} GroupMembership'.format(name)
+        retcode = __salt__['cmd.retcode'](cmd) == 0
+        for user in members_list.split(","):
```
Comment (cachedout) [inline salt/modules/mac_group.py]: Single-quotes here, please.

**Comment by cachedout** [comment]
@adambarnett52 Apologies for the ticky-tack item, but we normally prefer single-quotes to double-quotes. If you wouldn't mind changing that we'll get this in right away. The rest looks great and this will be an excellent addition to Salt!

**Comment by adambarnett52** [comment]
That is understandable, I'm out of the office for a few days, but will do it ASAP on my return.

**Comment by jfindlay** [comment]
@adambarnett52, no problem.  I've taken care of it, thanks.

**Pull request merged**